### DISCUSSION TITLE
Add a repository to GitHub APP using API

### DISCUSSION THREAD

**Discussion by ghost** [community/community#27280] [Apps, API and Webhooks]
Hello All, 
We have an organization, with an installed App and repositories owned by it. We want to use a tech user (not within the org admins) to add/remove a repository to/from the GitHub App. 

What permissions does the tech user need and what is the sequence of REST API calls?

Addition information
The tech user that we use to call the REST API that adds a repo to GITHub APP is the admin of the repository but not of the organization. The APP is installed in the organization and own by the organization.
We got a permission error when we try to call the REST API

**Comment by jfullstackdev** [comment]
> We got a permission error when we try to call the REST API

what permission error is that ?

**Reply by ghost** [reply]
Hello ,
The error message is "You do not have permission to modify this app on "the org." Please contact Organization Owner."
We are trying to use this API https://docs.github.com/en/rest/apps/installations#add-a-repository-to-an-app-installation.

**Reply by jfullstackdev** [reply]
is the project public ? maybe if we can see that, it will be better 

how about correct scopes ? removing and adding repos are write permissions

**Reply by ghost** [reply]
No, unfortunately, the repo is not public.

The tech user is the creator of the repository and then the repository admin, which written permission are we missing?

**Reply by jfullstackdev** [reply]
do you provide users with the admin's Personal Access Token ? it's quite a security issue but since you are just testing it, maybe you can tick all the scopes to see whether it will work

**Reply by ghost** [reply]
As you said, just for testing I added all the permissions to tech user Personal Access Token. Same error.

**Reply by iamfoysal** [reply]
you can see ,  https://github.com/iamfoysal/JSON-API-Functionality

**Reply by jfullstackdev** [reply]
> As you said, just for testing I added all the permissions to tech user Personal Access Token. Same error.

try giving the PAT to another user that is not part of the Org, but not for the App, rather for the write access at the Org, say creating a new repo, if that is not working too, then there is the problem

**Reply by ghost** [reply]
Hi, the token is already used for repository creation using the REST API. It works properly. If I understand correctly your suggestion.

**Reply by ghost** [reply]
According to the error `message": "You do not have permission to modify this app on <org>. Please contact an Organization Owner.`, is possible that the tech user should be the GitHun APP owner?  Or is not necessary in order to add a repository to gitHub APP?

**Reply by jfullstackdev** [reply]
what I'm seeing is, the PAT is from a collaborator ? but the owner rights are needed for that specific activity, I'm not very sure if the PAT of one owner will, but I think it will, because owner has certain rights that are exclusive for them only

**Reply by ghost** [reply]
Yes, the PAT is from a collaborator. The GitHub APP is installed and owned by the organization.

**Reply by ghost** [reply]
I tried to do two kinds of tests using the tech user with a PAT assigned to it. The tech user is a collaborator on the organization and has GitHub App managers permission on the organization.

- I try to add a repository (owned by the tech user) to a GItHub APP (owned by the tech user). This association works properly.
- I try to add a repository (owned by the organization) to a GitHub APP (owned by the organization). In this case, I got an error message": "You do not have permission to modify this app on <org>. Please contact an Organization Owner.` 

We don't want to transfer the ownership of the GitHub app to the tech user, we want to add a repository to GitHub App using this tech user. Is that possible?
